src/js/select2/data/maximumSelectionLength.js

```diff
@@ -0,0 +1,29 @@
+define([
+
+], function (){
+  function MaximumSelectionLength (decorated, $e, options) {
+    this.maximumSelectionLength = options.get('maximumSelectionLength');
+
+    decorated.call(this, $e, options);
+  }
+
+  MaximumSelectionLength.prototype.query =
+    function (decorated, params, callback) {
+
+    var count = this.$element.val() != null ? this.$element.val().length : 0;
```

Comment:
We might want to use `this.current()` here, to support cases where the current value is not actually stored in `$element.val()` (like the old hidden input, if we ever choose to support it again). This will also avoid the issue of it triggering an error if used on a non-multiple select (instead it just won't work, which more _slightly more_ sense).